kernel/src/process/posix_thread/mod.rs

```diff
@@ -141,21 +141,38 @@ impl PosixThread {
         &self.sig_mask
     }
 
+    /// Returns the pending signal of the thread.
+    ///
+    /// This method will also return the signals that are blocked or ignored.
     pub fn sig_pending(&self) -> SigSet {
-        self.sig_queues.sig_pending()
+        let process = self.process();
+        self.sig_queues.sig_pending() | process.sig_queues().sig_pending()
     }
 
     /// Returns whether the thread has some pending signals
-    /// that are not blocked.
+    /// that are not blocked and not ignored.
     pub fn has_pending(&self) -> bool {
+        let process = self.process();
+
+        // Fast path: No signals are pending.
```

Comment:
In fact, this "fast" path is not very fast. `self.process()` always creates an owned `Arc<Process>`, which we must drop later. If we have `Context` (e.g., in the task loop), we can do this better. But I'm not sure if we need to optimize this.